contracts/nft/io/src/lib.rs

```diff
@@ -0,0 +1,140 @@
+#![no_std]
+use gmeta::{In, InOut, Metadata};
+use gstd::{prelude::*, ActorId};
+
+pub type TokenId = u128;
+pub const ZERO_ID: ActorId = ActorId::zero();
+
+pub struct NftMetadata;
+
+impl Metadata for NftMetadata {
+    type Init = In<InitNft>;
+    type Handle = InOut<NftAction, NftEvent>;
+    type Reply = ();
+    type Others = ();
+    type Signal = ();
+    type State = InOut<StateQuery, StateReply>;
+}
+
+#[derive(Default, Debug, Encode, Decode, TypeInfo)]
+pub struct Config {
+    pub max_mint_count: Option<u128>,
+}
+
+#[derive(Default, Debug, Encode, Decode, TypeInfo)]
+pub struct InitNft {
+    pub collection: Collection,
+    pub config: Config,
+}
+
+#[derive(Default, Debug, Encode, Decode, TypeInfo)]
+pub struct Collection {
+    pub name: String,
+    pub description: String,
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub enum NftAction {
+    Mint {
+        to: ActorId,
+        token_metadata: TokenMetadata,
+    },
+    Burn {
+        token_id: TokenId,
+    },
+    Transfer {
+        to: ActorId,
+        token_id: TokenId,
+    },
+    Approve {
+        to: ActorId,
+        token_id: TokenId,
+    },
+    Owner {
```

Comment:
I'd add a verb to the action. `GetOwner` or `GetOwnerOf`.
```suggestion
    GetOwner {
```